internal/storage/append_batch.go

```diff
@@ -0,0 +1,71 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/cockroachdb/pebble"
+
+	"github.com/kakao/varlog/pkg/types"
+)
+
+var appendBatchPool = sync.Pool{
+	New: func() interface{} {
+		return &AppendBatch{
+			dk: make([]byte, dataKeyLength),
+			ck: make([]byte, commitKeyLength),
+			cc: make([]byte, commitContextLength),
+		}
+	},
+}
+
+// AppendBatch is a batch to put one or more log entries.
+type AppendBatch struct {
+	batch     *pebble.Batch
+	writeOpts *pebble.WriteOptions
+	dk        []byte
+	ck        []byte
+	cc        []byte
+}
+
+func newAppendBatch(batch *pebble.Batch, writeOpts *pebble.WriteOptions) *AppendBatch {
+	ab := appendBatchPool.Get().(*AppendBatch)
+	ab.batch = batch
+	ab.writeOpts = writeOpts
+	return ab
+}
+
+func (ab *AppendBatch) release() {
+	ab.batch = nil
+	ab.writeOpts = nil
+	appendBatchPool.Put(ab)
+}
+
+// SetLogEntry inserts a log entry.
+func (ab *AppendBatch) SetLogEntry(llsn types.LLSN, glsn types.GLSN, data []byte) error {
```

Comment:
In what cases are data and glsn stored at the same time? Is AppendBatch only for replication?

Reply:
AppendBatch will be used only for synchronization.